lib/v2/screens/sign_up/display_name/display_name.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:seeds/i18n/edit_name.i18n.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/text_form_field_custom.dart';
+import 'package:seeds/v2/design/app_theme.dart';
+import 'package:seeds/v2/screens/sign_up/viewmodels/bloc.dart';
+
+class DisplayName extends StatefulWidget {
+  @override
+  _DisplayNameState createState() => _DisplayNameState();
+}
+
+class _DisplayNameState extends State<DisplayName> {
+  late SignupBloc _bloc;
+  final _keyController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    _bloc = BlocProvider.of<SignupBloc>(context);
+    _keyController.text = _bloc.state.displayNameState.displayName ?? '';
+    _keyController.addListener(() {
+      setState(() {
+        // Do nothing
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    _keyController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: _navigateBack,
+      child: Scaffold(
+        appBar: AppBar(),
+        body: BlocConsumer<SignupBloc, SignupState>(
```

Comment:
BlocConsumer vs BlocBuilder


`BlocConsumer`
BlocConsumer exposes a builder and listener in order react to new states. BlocConsumer is analogous to a nested BlocListener and BlocBuilder but reduces the amount of boilerplate needed. `BlocConsumer should only be used when it is necessary to both rebuild UI and execute other reactions to state changes in the bloc`

Since we are not using the listener i think we should be using the BlocBuilder here.

Reply:
I used it just in case we need to listen to events but we can always add it when needed. Changed it to builder.